evm_loader/lib/src/account_storage.rs

```diff
@@ -2597,20 +2575,11 @@ mod tests {
 
         let code = hex!("13412971").to_vec();
         let contract = &ACTUAL_SUICIDE;
-        // TODO: Should we deploy new contract by the previous address?
-        assert_eq!(
-            storage
-                .set_code(contract.address, LEGACY_CHAIN_ID, code.clone())
-                .await
-                .is_ok(),
-            true,
-        );
-        storage.verify_used_accounts(&[(fixture.contract_pubkey(contract.address), true, false)]);
-        storage.verify_upgrade_rent(0, 0);
-        storage.verify_regular_rent(
-            fixture.contract_rent(&code),
-            fixture.contract_rent(contract.code),
-        );
+
+        assert!(storage
+            .set_code(contract.address, LEGACY_CHAIN_ID, code.clone())
+            .await
+            .is_err());
```

Comment:
It's better to check the error like in https://github.com/neonlabsorg/neon-evm/blob/4bf0f0c400deade95d36d77f6f2c6180576abe16/evm_loader/lib/src/account_storage.rs#L2506